flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/ExpressionReducer.scala

```diff
@@ -78,6 +79,10 @@ class ExpressionReducer(
            (SqlTypeName.MAP, _) |
            (SqlTypeName.MULTISET, _) => None
 
+      // Exclude some JSON functions which behave differently when called as an argument of another
```

Comment:
My understanding from this comment is that the JSON_OBJECT function will just skip constant folding right?

Reply:
Yes, that's it. :-) We give this function some funky behavior to make it more useful, but in consequence we cannot have it be reduced.